core/vm/celo_contracts.go

```diff
@@ -46,7 +47,7 @@ func celoPrecompileAddress(index byte) common.Address {
 }
 
 func (ctx *celoPrecompileContext) IsCallerGoldToken() (bool, error) {
-	return true, nil // TODO
+	return addresses.GoldTokenAddress == ctx.caller, nil
```

Comment:
Worth checking if we need to compare properly.

Reply:
Seems to be safe since it's addresses and not address pointers.